sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1759,13 +1759,76 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_tuple_at_codegen(self, row, col):
+    """
+    Example of generated implementation:
+        def _df_getitem_tuple_at_impl(self, idx):
+            row, _ = idx
+            check_row = False
+            for i in prange(len(self._dataframe.index)):
+                if self._dataframe.index[i] == row:
+                    check_row = True
+            if check_row:
+                data = self._dataframe._data[2]
+                res_data = pandas.Series(data, index=self._dataframe.index)
+                return res_data.at[row]
+            raise IndexingError('Index is out of bounds for axis')
+    """
+    func_lines = ['def _df_getitem_tuple_at_impl(self, idx):']
+    check_col = False
+    for i in range(len(self.columns)):
+        if self.columns[i] == col:
+            check_col = True
+            col_idx = i
+    if check_col == True:  # noqa
+        func_lines += ['  row, _ = idx',
+                       '  check_row = False',
+                       '  for i in prange(len(self._dataframe.index)):',
+                       '    if self._dataframe.index[i] == row:',
+                       '      check_row = True',
+                       '  if check_row:',
+                       f'    data = self._dataframe._data[{col_idx}]',
+                       '    res_data = pandas.Series(data, index=self._dataframe.index)',
+                       '    return res_data.at[row]',
+                       "  raise IndexingError('Index is out of bounds for axis')"]
+
+    else:
+        raise IndexingError('Index is out of bounds for axis')
```

Comment:
I propose to change the code a bit
```suggestion
    for i in range(len(self.columns)):
        if self.columns[i] == col:
            func_lines += [
                '  row, _ = idx',
                '  check_row = False',
                '  for i in prange(len(self._dataframe.index)):',
                '    if self._dataframe.index[i] == row:',
                '      check_row = True',
                '  if check_row:',
                f'    data = self._dataframe._data[{i}]',
                '    res_data = pandas.Series(data, index=self._dataframe.index)',
                '    return res_data.at[row]',
                '  raise IndexingError("Index is out of bounds for axis")'
            ]
            break
    else:
        raise IndexingError('Index is out of bounds for axis')
```